web/modules/custom/joinup_notification/src/EntityMessageHelperInterface.php

```diff
@@ -0,0 +1,96 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace Drupal\joinup_notification;
+
+use Drupal\Core\Entity\EntityInterface;
+use Drupal\Core\Entity\EntityStorageInterface;
+use Drupal\message\MessageInterface;
+
+/**
+ * Interface for services that assist in managing messages for entities.
+ */
+interface EntityMessageHelperInterface {
+
+  /**
+   * Constant representing sorting by ascending creation date.
+   */
+  const SORT_ASC = 'ASC';
+
+  /**
+   * Constant representing sorting by descending creation date.
+   */
+  const SORT_DESC = 'DESC';
+
+  /**
+   * Creates a new message that is associated with the given entity.
+   *
+   * @param \Drupal\Core\Entity\EntityInterface $entity
+   *   The entity that will be referenced in the new message.
+   * @param string $template
+   *   The message template ID.
+   * @param array $arguments
+   *   The array of arguments that will be used to replace token-like strings in
+   *   the message.
+   * @param string $field_name
+   *   The name of the field that references the entity for which the message is
+   *   to be created.
+   *
+   * @return \Drupal\message\MessageInterface
+   *   The newly created, unsaved, message.
+   *
+   * @throws \InvalidArgumentException
+   *   Thrown when the passed entity cannot be referenced. Possibly since it
+   *   hasn't yet been saved and does not have an ID yet.
+   */
+  public function createMessage(EntityInterface $entity, string $template, array $arguments, string $field_name): MessageInterface;
+
+  /**
+   * Returns the messages that are associated with the given entity.
+   *
+   * @param \Drupal\Core\Entity\EntityInterface $entity
+   *   The entity that is referenced by the message.
+   * @param string $template
+   *   The message template ID.
+   * @param string $field_name
+   *   The name of the field that references the entity for which the message is
+   *   to be created.
+   * @param array $values
+   *   An associative array of properties to filter the messages by, where the
+   *   keys are the property names and the values are the values those
+   *   properties must have.
+   * @param int $limit
+   *   The maximum number of entities to return. Defaults to 10 messages.
+   * @param string $order
+   *   Whether to sort by ascending or descending creation date. Defaults to
+   *   descending order.
+   *
+   * @return \Drupal\Core\Entity\EntityInterface[]
+   *   The corresponding messages.
+   */
+  public function getMessages(EntityInterface $entity, string $template, string $field_name, array $values = [], int $limit = 10, string $order = self::SORT_DESC): array;
```

Comment:
The `$order` here is never changed in our codebase and we only use the `DESC` order to sort the messages. I don't think that it is needed as a parameter which also makes the constants uneeded, especially the `SORT_ASC` which is never used. I know it is normal that we have it because we have the `SORT_DESC` one too, but these are never used so the parameter is not needed at all.

Reply:
But then how are you going to retrieve a list of messages in ascending order if you need it?

This is a generic service to retrieve any kind of message related to any kind of entity, it is not limited to the use case of this particular issue, or I would have called it the `OgRoleChangeNotificationHelperInterface`.